### DISCUSSION TITLE
Can't get a reusable workflow on a branch executed

### DISCUSSION THREAD

**Discussion by sgandon** [community/community#65598] [Actions]
### Select Topic Area

Bug

### Body

I am modifying a reusable workflow and created a branch (foo) for that matter.
When I try to test this workflow by calling it the main branch is used and not the branch `foo`.

in order to call my workflow I simply use the recommanded branch name  from the doc like this
```
uses: myaccount/myrepo/.github/workflows/reusable_workflow.yaml@foo
```

but this fails and launches the one on master.
Any idea why ?

**Comment by arafajs** [comment]
To trigger the workflow from the foo branch, you need to use the on keyword to specify the branch that the workflow should be triggered on. For example:

```
on:
  workflow_dispatch:
    branches:
      - foo
```

**Reply by sgandon** [reply]
Thanks for your answer but this is not my question.
I don't want to trigger a workflow from a branch I want to call a reusable workflow hosted in a another repo and in a specific branch.
